### PR TITLE
feat(math): implement pasta curves

### PR DESCRIPTION
**Pull request by Insun35**
Pasta curves are the two curves; pallas and vesta. Each curve's scalar field is the other curve's base field, which is practical for recursion.

See https://o1-labs.github.io/proof-systems/specs/pasta.html

This PR should be merged after #335

### CODE REVIEW

File: docs/how_to_contribute/math/how_to_add_elliptic_curves.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# How to add elliptic curves
+
+Follow this guide to add a new elliptic curve for Tachyon.
+
+_Note_: We have our own development conventions. Please read the [conventions docs](/docs/how_to_contribute/conventions.md) before contributing.
+
+## Add a `BUILD.bazel` file
+
+Begin by creating a directory named `<new_elliptic_curve>` in `/tachyon/math/elliptic_curves/`. Add a `BUILD.bazel` into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the elliptic curve code based on these parameters when it builds the target.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/how_to_add_elliptic_curves.md:9]:
```suggestion
Begin by creating a directory named `<new_elliptic_curve>` in `/tachyon/math/elliptic_curves/`. Add a `BUILD.bazel` file into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the elliptic curve code based on these parameters when it builds the target.

File: docs/how_to_contribute/math/how_to_add_elliptic_curves.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+First, implement the Scalar Field and Base Field on which the Elliptic curve is defined. See [here](/docs/how_to_contribute/math/how_to_add_prime_field.md) for how to add a Prime Field.
+
+After implementing the Fields, by inputting the parameters of the elliptic curve into generator(`generate_ec_points`) as shown below:
```
Comment (ashjeong) [inline docs/how_to_contribute/math/how_to_add_elliptic_curves.md:13]: What does this mean?

File: docs/how_to_contribute/math/how_to_add_elliptic_curves.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Follow this guide to add a new elliptic curve for Tachyon.
+
+_Note_: We have our own development conventions. Please read the [conventions docs](/docs/how_to_contribute/conventions.md) before contributing.
+
+## Add a `BUILD.bazel` file
+
+Begin by creating a directory named `<new_elliptic_curve>` in `/tachyon/math/elliptic_curves/`. Add a `BUILD.bazel` into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the elliptic curve code based on these parameters when it builds the target.
+
+First, implement the Scalar Field and Base Field on which the Elliptic curve is defined. See [here](/docs/how_to_contribute/math/how_to_add_prime_field.md) for how to add a Prime Field.
```
Comment (ashjeong) [inline docs/how_to_contribute/math/how_to_add_elliptic_curves.md:11]:
```suggestion
First, implement the _scalar field_ and _base field_ on which the elliptic curve is defined. See [here](/docs/how_to_contribute/math/how_to_add_prime_field.md) for how to add a _prime field_.

File: docs/how_to_contribute/math/how_to_add_elliptic_curves.md (hunk 2 above)
Comment (ashjeong) [reply docs/how_to_contribute/math/how_to_add_elliptic_curves.md:13]: is it like 
```suggestion
After implementing the Fields, input the parameters of the elliptic curve into generator(`generate_ec_points`) like shown below:

Comment (dongchangYoo) [reply docs/how_to_contribute/math/how_to_add_elliptic_curves.md:13]: It seems like typo.

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: docs/how_to_contribute/math/how_to_add_elliptic_curves.md
Comment (ashjeong) [inline docs/how_to_contribute/math/how_to_add_elliptic_curves.md:1]: maybe some sort of ending like "The elliptic curve is now ready to be used in Tachyon!" would be good

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**